src/types.js

```diff
@@ -2,12 +2,17 @@
 import Jss from './Jss'
 import StyleSheet from './StyleSheet'
 import ConditionalRule from './plugins/ConditionalRule'
+import RegularRule from './plugins/RegularRule'
 
 export type ToCssOptions = {
   indent?: number
 }
 
-export type generateClassName = (str: string, rule: Rule) => string
+export type generateClassName = (rule: Rule) => string
+
+// TODO
+// Find a way to declare all types: Object|string|Array<Object>
+export type JssStyle = Object
```

Comment:
I narrowed down the problem in [this tweet](https://twitter.com/oleg008/status/844634629718102017)

https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoALgTwA4FMwDKWM+AvGAPIBGAVrgMboY74AiDMY5RmJAPgEEATkICGmADw8SAPlRgwqKAFcAdowCWcVWAAWo1QBMS03AAoAzsVwAuQtYCUYAN6oAvqiVrN2vQeO4wmKYZoYcduz0ME6uChpQYGZB4gB0GhbJIWFRDk5CuOjKQjrOHh4KXuroWjr6RiSh4WCR0S7yYPTaVmD56Fx+9YEi4o057fGJvXkFRTq97b2zAwGmo9HuQA